emscripten.py

```diff
@@ -991,20 +991,20 @@ def make_function_tables_impls(function_table_data):
     args = ','.join(['a' + str(i) for i in range(1, len(sig))])
     arg_coercions = ' '.join(['a' + str(i) + '=' + shared.JS.make_coercion('a' + str(i), sig[i]) + ';' for i in range(1, len(sig))])
     coerced_args = ','.join([shared.JS.make_coercion('a' + str(i), sig[i]) for i in range(1, len(sig))])
-    sig_mask = str(table.count(','))
-    ret = ('return ' if sig[0] != 'v' else '') + shared.JS.make_coercion('FUNCTION_TABLE_%s[index&%s](%s)' % (sig, sig_mask, coerced_args), sig[0])
-    if not shared.Settings.EMULATED_FUNCTION_POINTERS:
-      function_tables_impls.append('''
+    if shared.Settings.EMULATED_FUNCTION_POINTERS:
```

Comment:
I think this part of the patch could be simplified by setting `sig` to `X` if `EMULATED_FUNCTION_POINTERS`? That would avoid duplicated code in the if-else here (the return handling, the `make_coercion` call, etc.

Possibly we need a notion of an `effective_sig` if just changing the `sig` itself here or higher up does not make sense.